### PR TITLE
fix(multiple database): all database type must be same in multiple databases task

### LINKED ISSUE #3481: [Bug]: When distributing multi-database changes, there are no restrictions on different types of databases.
### ODC version

ODC 4.3.2

### OB version

No related.

### What happened?

I can select obmysql and Oracle data sources to distribute multi-database change tasks simultaneously.
image

### What did you expect to happen?

You can only select one type of database to distribute multi-database change tasks.

### How can we reproduce it (as minimally and precisely as possible)?

Follow what happened.

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #3481

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/databasechange/MultipleDatabaseChangePreprocessor.java [hunk 1]
```diff
@@ -79,6 +79,10 @@ public void process(CreateFlowInstanceReq req) {
             throw new BadArgumentException(ErrorCodes.BadArgument,
                     String.format("All databases must belong to the same project: %s", project.getName()));
         }
+        if (databases.stream().map(x -> x.getDataSource().getType()).distinct().count() != 1) {
+            throw new BadArgumentException(ErrorCodes.BadArgument,
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/databasechange/MultipleDatabaseChangePreprocessor.java:83]: BadRequset, not BadArgument

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/databasechange/MultipleDatabaseChangePreprocessor.java]: done

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**